registry/config.go

```diff
@@ -302,34 +315,22 @@ func splitReposName(reposName string) (string, string) {
 
 // NewRepositoryInfo validates and breaks down a repository name into a RepositoryInfo
 func (config *ServiceConfig) NewRepositoryInfo(reposName string, bySearch bool) (*RepositoryInfo, error) {
```

Comment:
Nit: For the sake of code reuse/readability, is it possible to restructure NewRepositoryInfo to take a bool arg? This arg would indicate call from NormalizeLocalName. There are only 3 callers for NewRepositoryInfo, so changing the function signature would be okay. Meanwhile, in NewRepositoryInfo, since most of the code that we need is here, we can use the bool flag to navigate our way around NewIndexInfo (which causes the network lookup).